doc/templates/page.html

```diff
@@ -0,0 +1,6 @@
+{%- extends "basic/page.html" %}
+
+{%- block extrahead %}
+{{ super() }}
+<meta name="docsearch:version" content="{{ version }}"/>
```

Comment:
## Versioning

**Problem:** Searching in any version will currently link to results in the stable version.

**Possible solution:** Use meta tags and facet filters. See https://github.com/scikit-learn/scikit-learn/pull/29138#issuecomment-2145887567, [meta tags docs](https://docsearch.algolia.com/docs/required-configuration#introduce-global-information-as-meta-tags), and [facet filter docs](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/). I've opened https://github.com/pydata/pydata-sphinx-theme/issues/1951 upstream to add the meta tag, and it is also easy to tweak the `page.html` template as a temporary workaround.